text/0013-client-side-controls.md

```diff
@@ -0,0 +1,70 @@
+# Client-side window controls
+
+## Summary
+
+Enables applications with client-side decorations to maximize, minimize, and initiate a window drag with the mouse.
+
+## Motivation
+
+GUI toolkits implementing client-side decorations need to implement a headerbar widget which has access to triggering window methods to maximize, minimize, and drag the window with the mouse.
+
+## Guide-level explanation
+
+An application will be able define messages for window close, drag, minimize, and maximize, then call upon these window commands from the update method.
+
+```rs
+#[derive(Debug, Clone)]
+pub enum Message {
+    Close,
+    Drag,
+    Minimize,
+    Maximize,
+}
+
+fn update(&mut self, message: Message) -> iced::Command<Self::Message> {
+    match message 
+        Message::Close => self.should_exit = true,
+        Message::Drag => return iced_native::window::drag(),
+        Message::Minimize => return iced_native::window::minimize(),
+        Message::Maximize => return iced_native::window::toggle_maximize(),
+        ...
+    }
```

Comment:
Quite elegant!

This design actually seems to expose a lack of consistency with `should_exit`. I believe we should eventually implement a `Command` for `window::close` and remove `should_exit` altogether.

But no need to reflect that in this RFC!